pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -235,33 +288,74 @@ func nodesWherePreemptionMightHelp(nodes []*framework.NodeInfo, m framework.Node
 	return potentialNodes
 }
 
+type candidateList struct {
+	idx   int32
+	items []Candidate
+}
+
+func newCandidateList(size int32) *candidateList {
+	return &candidateList{idx: -1, items: make([]Candidate, size)}
+}
+
+// add adds a new candidate to the internal array atomically.
+func (cl *candidateList) add(c *candidate) {
+	if idx := atomic.AddInt32(&cl.idx, 1); idx < int32(len(cl.items)) {
+		cl.items[idx] = c
+	}
+}
+
+// size returns the number of candidate stored atomically.
+func (cl *candidateList) size() int32 {
```

Comment:
size() is also not atomic and assumes all add() operations have been completed

Reply:
True, but we also don't access the list elements until all add() operations complete. I've updated the function doc to reflect this more accurately.